sdl_android/src/main/java/com/smartdevicelink/protocol/SdlPacket.java

```diff
@@ -344,6 +353,22 @@ public SdlPacket createFromParcel(Parcel in) {
 		}
 
     };
-	
-	
+
+	public void putTag(String tag, Object data){
+		bsonPayload.put(tag, data);
```

Comment:
This is where you should instantiate the `bsonPayload` object `if(bsonPayload==null){bsonPayload = new HashMap<String,Object>());}`